pkg/provider/provider.go

```diff
@@ -1,3 +1,22 @@
+// Package Provider provides the interface for providers which satisfy the [compiler.ProviderPlugin].
```

Comment:
```suggestion
// Package provider provides the interface for providers which satisfy the [compiler.ProviderPlugin].
```

(Right `provider` is the package, `Provider` is an interface within that package.)